### PR TITLE
Leaves on trees decay when logs burned by fire. Completes Vanilla-399

### PR DESCRIPTION
**Pull request by Amperial**
Need to fix the formatting. Still getting used to IntelliJ IDEA (used eclipse before)

### CODE REVIEW

**Comment by Amperial** [comment]
Formatting fixed, changes tested and confirmed working.

File: src/main/java/org/spout/vanilla/material/block/misc/Fire.java [hunk 1]
```diff
@@ -182,8 +182,12 @@ public void onDynamicUpdate(Block b, long updateTime, int data) {
 				// Destroy the old block
 				if (mat instanceof TntBlock) {
 					((TntBlock) mat).onIgnite(sBlock, toCause(b)); // Ignite TntBlock
+				} else if (sBlock instanceof VanillaBlockMaterial) {
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/material/block/misc/Fire.java]: `sBlock` is a `Block` which will never be a `VanillaBlockMaterial`. Replace `sBlock` with `mat`.

File: src/main/java/org/spout/vanilla/material/block/misc/Fire.java [hunk 2]
```diff
@@ -182,8 +182,12 @@ public void onDynamicUpdate(Block b, long updateTime, int data) {
...
+					VanillaBlockMaterial sVanillaBlock = (VanillaBlockMaterial) sBlock; // Destroy vanilla blocks without drops
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/material/block/misc/Fire.java]: Same thing here `sBlock` -> `mat`

**Comment by Amperial** [comment]
That means it was still setting the logs to air, and the leaves actually burning instead of decaying..
Odd that it appeared as if they were decaying, since some of the leaves never actually got set on fire :P

**Comment by lukespragg** [comment]
All good on this @Sleaker / @Zidane?

**Comment by kitskub** [comment]
@ampayne2 the fact that it was "confirmed working" when the code was wrong makes me suspicious as to if this is really the correct "fix" or if it was even broken to begin with.

**Comment by Amperial** [comment]
@kitskub it was confirmed working as far as I could tell, which involved burning a forest of trees and crashing every time an item dropped..
It was definitely broken to begin with because the leaf decay task is scheduled on log destroy, which setMaterial doesn't trigger. If getDrops().clear() on the VanillaBlockMaterial of the log works as I believe it does, the fix should work fine although i'm not completely sure since I didn't write the code for drops.
Making physics and item dropping more stable is kinda high priority so it's possible to fully test changes; testing this is hard enough, but testing my work on getting both iron and wooden doors to drop is impossible because they don't attempt to drop without physics enabled (which freezes the server).

**Comment by kitskub** [comment]
@Sleaker @Zidane someone review this please

**Pull request merged**